administrator/components/com_templates/src/Controller/TemplateController.php

```diff
@@ -980,4 +984,105 @@ public function ajax()
 
 		$app->close();
 	}
+
+
+	/**
+	 * Method for creating a child template.
+	 *
+	 * @return  boolean     true on success, false otherwise
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function child()
+	{
+		// Check for request forgeries
+		$this->checkToken();
+
+		$app = $this->app;
+		$this->input->set('installtype', 'folder');
+		$newNameRaw = $this->input->get('new_name', null, 'string');
+		// Only accept letters, numbers and underscore for template name
+		$newName    = preg_replace('/[^a-zA-Z0-9_]/', '', $newNameRaw);
+		$templateID = (int) $this->input->getInt('id', 0);
+		$file       = (string) $this->input->get('file', '', 'cmd');
+
+		// Access check.
+		if (!$this->allowEdit())
+		{
+			$app->enqueueMessage(Text::_('JLIB_APPLICATION_ERROR_SAVE_NOT_PERMITTED'), 'error');
+
+			return false;
+		}
+
+		$this->setRedirect('index.php?option=com_templates&view=template&id=' . $templateID . '&file=' . $file);
+
+		/* @var \Joomla\Component\Templates\Administrator\Model\TemplateModel $model */
+		$model = $this->getModel('Template', 'Administrator');
+		$model->setState('new_name', $newName);
+		$model->setState('tmp_prefix', uniqid('template_child_'));
+		$model->setState('to_path', $app->get('tmp_path') . '/' . $model->getState('tmp_prefix'));
+
+		// Process only if we have a new name entered
+		if (!strlen($newName)) {
+			$this->setMessage(Text::sprintf('COM_TEMPLATES_ERROR_INVALID_TEMPLATE_NAME'), 'error');
+
+			return false;
+		}
+
+		// Process only if user is allowed to create child template
+		if (!$this->app->getIdentity()->authorise('core.create', 'com_templates')) {
+			$this->setMessage(Text::_('COM_TEMPLATES_ERROR_CREATE_NOT_PERMITTED'), 'error');
+
+			return false;
+		}
+
+		// Check that new name is valid
+		if (($newNameRaw !== null) && ($newName !== $newNameRaw)) {
+			$this->setMessage(Text::_('COM_TEMPLATES_ERROR_INVALID_TEMPLATE_NAME'), 'error');
+
+			return false;
+		}
+
+		// Check that new name doesn't already exist
+		if (!$model->checkNewName()) {
+			$this->setMessage(Text::_('COM_TEMPLATES_ERROR_DUPLICATE_TEMPLATE_NAME'), 'error');
+
+			return false;
+		}
+
+		// Check that from name does exist and get the folder name
+		$fromName = $model->getFromName();
+
+		if (!$fromName) {
+			$this->setMessage(Text::_('COM_TEMPLATES_ERROR_INVALID_FROM_NAME'), 'error');
+
+			return false;
+		}
```

Comment:
All of the `if` statements need opening brace on their own line

```suggestion
		if (!$fromName) 
		{
			$this->setMessage(Text::_('COM_TEMPLATES_ERROR_INVALID_FROM_NAME'), 'error');

			return false;
		}
```